src/common_dims.cpp

```diff
@@ -152,5 +146,42 @@ common_dims common_dims::compute(const std::vector<std::size_t>& dims1,
     return cd;
 }
 
+const std::vector<std::vector<std::size_t>>* common_dims::get_axes_map(std::size_t n) const
```

Comment:
can you add docstring on what this is supposed to do ? It seems strange from looking at logic that it return either axes_map1 or map2 just on the basis of size. 

Also requires unit-tests inside `test/common_dims.cpp`

Reply:
When we are computing a common dimension we have two shapes that are usually different ranks this lets us get the axes map without needing to know whether it was passed as the first or second parameter.

I guess there could be a case where it expands and collpases different parts of the tensor(ie `{2, 32, 4}` -> `{64, 2, 2}` resulting in two shapes that have the same rank. We should bail on that case, let me add a test case for that.